waspc/data/Generator/templates/server/src/auth/providers/email/requestPasswordReset.ts

```diff
@@ -24,37 +27,48 @@ export function getRequestPasswordResetRoute({
     return async function requestPasswordReset(
         req: Request<{ email: string; }>,
         res: Response,
-    ): Promise<Response<{ success: true } | { success: false; message: string }>> {
-        const args = req.body || {};
+    ): Promise<Response<{ success: true }> | void> {
```

Comment:
![Screenshot 2023-12-29 at 13 15 26](https://github.com/wasp-lang/wasp/assets/2223680/e4db092d-381a-4439-aff9-2f499a76ad74)

I've seen this error and then I decided to add the `void` return value since I assumed as well it's connected to the `throw` statements.

But was a mistake on my part since `res.json` returns a response in Express **but** you don't have to `return res.json` in order for it to work properly. 

Why is this important? Because in one of the functions, I didn't have a `return` for each of the `res.json` calls which confused Typescript and then it demanded I modify the return type. It worked for Express, but Typescript wasn't happy.

After returning after each `res.json` the `void` return type can be safely removed.

I'll modify the code to reflect this new understanding 👍

Reply:
Makes sense, good catch!